src/tokenizers/strspn.h

```diff
@@ -4,17 +4,28 @@
 #include <string.h>
 
 namespace Simhash {
+    static const char* alphas = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+    struct StringToken {
+        const char* start;
+        size_t      length;
+    };
+
     struct Strspn {
         /**
-         * Return a pointer to the end of the next token or NULL if
-         * the end of string has been reached.
+         * Sets fields of the `output` to the first character of the
+         * token and its length, skipping preceding non-token characters.
+         * If the `string` does not contain any token (empty or constists
+         * of separators only), `output.start` is set to NULL, and
+         * `output.length` is set to 0.
+         *
+         * Note that for valid token length is never 0.
          */
-        const char* operator()(const char* last) const {
-            size_t s = strspn(last,
-                "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ");
-            return (*last != '\0') ? last + s : NULL;
+        void operator()(const char* string, StringToken& output) const {
+            const char* start = strpbrk(string, alphas);
+            output.start = start;
+            output.length = (start? strspn(start, alphas): 0);
```

Comment:
Martin's been working on configuring a style rule checker for us, but I don't think it's ready at this point.